concordium-consensus/tests/consensus/ConcordiumTests/LeaderElectionTest.hs

```diff
@@ -0,0 +1,145 @@
+                        predictLeadershipElectionNonce ss 6 (Just 7) 8 `shouldBe` Just [computeLeadershipElectionNonce ss 8],
+                        predictLeadershipElectionNonce ss 7 Nothing 15 `shouldBe` Just [computeLeadershipElectionNonce ss 15, computeLeadershipElectionNonce ss' 15],
+                        predictLeadershipElectionNonce ss 7 (Just 8) 15 `shouldBe` Just [computeLeadershipElectionNonce ss 15],
+                        predictLeadershipElectionNonce ss 7 (Just 11) 15 `shouldBe` Just [computeLeadershipElectionNonce ss' 15],
+                        maybeSublistExpectation (predictLeadershipElectionNonce ss 7 Nothing 15) (predictLeadershipElectionNonce ss' 10 Nothing 15),
+                        predictLeadershipElectionNonce ss 6 (Just 8) 15 `shouldBe` predictLeadershipElectionNonce ss'' 7 (Just 8) 15,
+                        predictLeadershipElectionNonce ss 6 Nothing 15 `shouldBe` predictLeadershipElectionNonce ss'' 7 Nothing 15,
+                        predictLeadershipElectionNonce ss 6 Nothing 8 `shouldBe` predictLeadershipElectionNonce ss'' 7 Nothing 8,
+                        predictLeadershipElectionNonce ss 7 (Just 11) 15 `shouldBe` predictLeadershipElectionNonce ss' 10 (Just 11) 15,
+                        predictLeadershipElectionNonce ss 7 (Just 8) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 7 (Just 11) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 7 Nothing 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 8) 15 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 11) 15 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 Nothing 15 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 8) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 11) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 Nothing 20 `shouldBe` Nothing
+                    ]
+    where
+        ss = SeedState {
+            epochLength = 9,
+            epoch = 0,
+            currentLeadershipElectionNonce = H.hash "1",
+            updatedNonce = H.hash "2"
+        }
+        vrfKP = fst $ VRF.randomKeyPair (mkStdGen 0)
+        bn = VRF.prove vrfKP "3"
+        ss' = updateSeedState 10 bn ss
+        ss'' = updateSeedState 7 bn ss
+
+genSeedstate :: Gen SeedState
+genSeedstate =
+    do
+        (el :: Word64) <- (*9) <$> choose (1,10000)
+        epoch <- choose (0, 1000)
+        return SeedState {
+            epochLength = fromIntegral el,
+            currentLeadershipElectionNonce = H.hash "1",
+            updatedNonce = H.hash "2",
+            ..
+        }
+
+-- Tests that both Maybe List's are Just and that former contains the latter as a sublist
```

Comment:
You check `IsInfixOf`, not that it is a sublist. Perhaps update the documentation.